k8s/upgrades/0.9.0-1.0.0/jiva/jiva_volume_upgrade.sh

```diff
@@ -0,0 +1,177 @@
+r_name=$(kubectl get deploy -n $ns $r_dep -o jsonpath="{range .spec.template.spec.containers[*]}{@.name}{'\n'}{end}" | grep "rep-con")
+
+# Fetch the older target and replica - ReplicaSet objects which need to be
+# deleted before upgrading. If not deleted, the new pods will be stuck in
+# creating state - due to affinity rules.
+
+c_rs=$(kubectl get rs -o name --namespace $ns -l openebs.io/persistent-volume=$pv,openebs.io/controller=jiva-controller | cut -d '/' -f 2)
+r_rs=$(kubectl get rs -o name --namespace $ns -l openebs.io/persistent-volume=$pv,openebs.io/replica=jiva-replica | cut -d '/' -f 2)
+
+################################################################
+# STEP: Update patch files with appropriate resource names     #
+#                                                              #
+# NOTES: Placeholder for resourcename in the patch files are   #
+# replaced with respective values derived from the PV in the   #
+# previous step                                                #
+################################################################
+
+# Check if openebs resources exist and provisioned version is 0.9.0
+
+if [[ -z $c_rs ]]; then
+    echo "Target Replica set not found"; exit 1;
+fi
+
+if [[ -z $r_rs ]]; then
+    echo "Replica Replica set not found"; exit 1;
+fi
+
+if [[ -z $c_dep ]]; then
+    echo "Target deployment not found"; exit 1;
+fi
+
+if [[ -z $r_dep ]]; then
+    echo "Replica deployment not found"; exit 1;
+fi
+
+if [[ -z $c_svc ]]; then
+    echo "Target service not found"; exit 1;
+fi
+
+if [[ -z $r_name ]]; then
+    echo "Replica container not found"; exit 1;
+fi
+
+if [[ -z $c_name ]]; then
+    echo "Target container not found"; exit 1;
+fi
+
+controller_version=`kubectl get deployment $c_dep -n $ns -o jsonpath='{.metadata.labels.openebs\.io/version}'`
+if [[ "$controller_version" != "$current_version" ]] && [[ "$controller_version" != "$target_upgrade_version" ]]; then
+    echo "Current Target deployment $c_dep version is not $current_version or $target_upgrade_version";exit 1;
+fi
+replica_version=`kubectl get deployment $r_dep -n $ns -o jsonpath='{.metadata.labels.openebs\.io/version}'`
+if [[ "$replica_version" != "$current_version" ]] && [[ "$replica_version" != "$target_upgrade_version" ]]; then
+    echo "Current Replica deployment $r_dep version is not $current_version or $target_upgrade_version";exit 1;
+fi
+
+controller_svc_version=`kubectl get svc $c_svc -n $ns -o jsonpath='{.metadata.labels.openebs\.io/version}'`
+if [[ "$controller_svc_version" != $current_version ]] && [[ "$controller_svc_version" != "$target_upgrade_version" ]] ; then
+    echo "Current Target service $c_svc version is not $current_version or $target_upgrade_version";exit 1;
+fi
+
+sed -u "s/@r_name@/$r_name/g" jiva-replica-patch.tpl.json | sed -u "s/@target_version@/$target_upgrade_version/g" | sed -u "s/@pv_name@/$pv/g" > jiva-replica-patch.json
```

Comment:
We need to indent the code and make this readable!!

Reply:
done